### ISSUE TITLE
Required CliOption

### ISSUE THREAD

**Issue by lookbusy1344** [dotmake-build/command-line#2]
Love the library! If I have a property like:

```
[CliOption(Description = "Description for Display", AllowedValues = new[] { "Big", "Small" }, Required = true)]
public string? Display { get; set; }
```

It correctly gives an error for:

```
prog.exe --display
prog.exe --display invalid
```

But it runs without error if the option is entirely missing:

```
prog.exe
```

Leaving the Display property as null.
Is this correct behaviour? And if so, how do I make the `--display` option mandatory?

**Comment by calacayir** [comment]
Normally `Required ` property is supposed to work but could it be about using nullable type `string?`. Maybe the combination with `AllowedValues` is causing the issue. I will check and let you know. By the way I just released v1.5 which supports binding to your custom types.

```c#
[CliOption(
    Description = "Description for Display", 
    AllowedValues = new[] { "Big", "Small" }, 
    Required = true
)]
public string Display { get; set; }
```

**Comment by lookbusy1344** [comment]
```
	[CliOption(Description = "Description for Display", Required = true)]
	public string Display { get; set; }
```

Still gives me the issue. This is on version 1.4

**Comment by calacayir** [comment]
FYI, this is now fixed in v1.5.2:

- Changed behaviour: Added Required property to CliArgumentAttribute, default is true for arguments.
Default is false for options (already had Required property). Whether you specify a default value for a property will
no longer effect the requirement of the argument or option. By contrast, when you set Required to true, the default value
of the property will be ignored.

When from now on, when you set Required property to true, it will be respected and error will be shown if it's missing.
